src/Microsoft.Azure.EventHubs/Primitives/ClientEntity.cs

```diff
@@ -55,6 +65,43 @@ public RetryPolicy RetryPolicy
         /// <returns>The asynchronous operation</returns>
         public abstract Task CloseAsync();
 
+        /// <summary>
+        /// Registers a <see cref="EventHubsPlugin"/> to be used with this client.
+        /// </summary>
+        public virtual void RegisterPlugin(EventHubsPlugin eventHubPlugin)
+        {
+            if (eventHubPlugin == null)
+            {
+                throw new ArgumentNullException(nameof(eventHubPlugin), Resources.ArgumentNullOrWhiteSpace.FormatForUser(nameof(eventHubPlugin)));
+            }
+            if (this.RegisteredPlugins.Any(p => p.Key == eventHubPlugin.Name))
+            {
+                throw new ArgumentException(nameof(eventHubPlugin), Resources.PluginAlreadyRegistered.FormatForUser(nameof(eventHubPlugin.Name)));
+            }
```

Comment:
You don't need this anymore since TryAdd does same job.

Reply:
Actually according this  : https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.tryadd?redirectedfrom=MSDN&view=netframework-4.7.2#System_Collections_Concurrent_ConcurrentDictionary_2_TryAdd__0__1

It throws `ArgumentNullException` Only if key is null which will never be the case because above is this code 

C# ```
            if (string.IsNullOrWhiteSpace(pluginName))
            {
                throw new ArgumentNullException(nameof(pluginName), Resources.ArgumentNullOrWhiteSpace.FormatForUser(nameof(pluginName)));
            }
```

Just my opinion but it's handy get a bit more information there about which one failed also.